squidb-addons/squidb-reactive/src/com/yahoo/squidb/reactive/ReactiveSquidDatabase.java

```diff
@@ -0,0 +1,138 @@
+                    "should never happen anyway.");
+        }
+    }
+
+    /**
+     * Create a new ReactiveSquidDatabase.
+     *
+     * @param context the Context, must not be null
+     */
+    public ReactiveSquidDatabase(Context context) {
+        super(context);
+        registerDataChangedNotifier(new PublishingDataChangedNotifier());
+    }
+
+    /**
+     * @param table the table to observe
+     * @return a new {@link Observable} that will be called whenever the given table is written to in a successful
+     * statement or transaction. The Observable will emit the table itself; to emit some other object like a Query, use
+     * {@link #observeTableForObject(SqlTable, Object)}
+     */
+    public <T extends SqlTable<?>> Observable<T> observeTable(T table) {
+        return observeTableForObject(table, table);
+    }
+
+    /**
+     * @param objectToEmit an object for the created Observable to emit
+     * @param table the table to observe
+     * @return a new {@link Observable} that will be called whenever the given table is written to in a successful
+     * statement or transaction. The Observable will emit the object passed in this method; for example, a Query to run.
+     */
+    public <T> Observable<T> observeTableForObject(final SqlTable<?> table, T objectToEmit) {
+        if (table == null) {
+            throw new IllegalArgumentException("Cannot observe a null table");
+        }
+
+        return observeForObject(objectToEmit, new Func1<Set<SqlTable<?>>, Boolean>() {
+            @Override
+            public Boolean call(Set<SqlTable<?>> changedTables) {
+                return changedTables.contains(table);
+            }
+        });
+    }
+
+    /**
+     * @param objectToEmit an object for the created Observable to emit
+     * @param tables the tables to observe
+     * @return a new {@link Observable} that will be called whenever any of the given tables are written to in a
+     * successful statement or transaction. The Observable will emit the object passed in this method; for example, a
+     * Query to run.
+     */
+    public <T> Observable<T> observeTablesForObject(final Collection<? extends SqlTable<?>> tables, T objectToEmit) {
+        return observeForObject(objectToEmit, new Func1<Set<SqlTable<?>>, Boolean>() {
+            @Override
+            public Boolean call(Set<SqlTable<?>> changedTables) {
+                for (SqlTable<?> table : tables) {
+                    if (changedTables.contains(table)) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+        });
+    }
+
+    private <T> Observable<T> observeForObject(final T objectToEmit, Func1<Set<SqlTable<?>>, Boolean> tableFilter) {
+        return changedTablePublisher
+                .filter(tableFilter)
+                .startWith(INITIAL_TABLE)
```

Comment:
It makes the first subscriber to this observable be immediately triggered. So for example, if you were emitting a Query you would immediately run the query and then re-run it on subsequent changes.